src/index.d.ts

```diff
@@ -54,6 +54,36 @@ export declare function styled<T extends keyof JSX.IntrinsicElements>(
 ) => ((props: P & JSX.IntrinsicElements[T]) => JSX.Element) & {
   className: (props: P & JSX.IntrinsicElements[T]) => string;
 };
+export declare function styled<T extends {}>(
+  component: (props: T) => JSX.Element
+): <P = T>(
+  args_0:
+    | string
+    | TemplateStringsArray
+    | CSSAttribute
+    | ((
+        props: P & {
+          theme?: any;
+          as?: string | number | symbol | undefined;
+          className?: any;
+          children?: any;
+        }
+      ) => string | CSSAttribute),
+  ...args_1: (
+    | string
+    | number
+    | ((
+        props: P & {
+          theme?: any;
+          as?: string | number | symbol | undefined;
+          className?: any;
+          children?: any;
+        }
+      ) => string | number | CSSAttribute | undefined)
+  )[]
```

Comment:
The types of the arguments are exactly the same as above.
Should I create new types and replace them?

Reply:
Yeah that's probably better. I very hastily ported these because Goobers types didn't play nice with ours but it was mostly copy and paste. I appreciate you taking the time to clean this up.